BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
 
 ## Library dependencies
 
+### Build Dependencies
+
 Binary builds additionally require the following packages to be
-available to the CMake build system.
+available.
 
 - Python 3.7 (development install)
```

Comment:
[consider] Linking to Python here, given we're linking to pybind?

Reply:
I'd actually linked all the dependencies, before undoing them with a "I don't wanna deal with discussions around that" thought.

Added them back in (Not system dependencies, that seemed inappropriate)
[a270dbb](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/a270dbb71d57199b1db2a80c2ca7abbba501ee71)